cluster-autoscaler/simulator/clustersnapshot/predicate/predicate_snapshot.go

```diff
@@ -101,6 +170,30 @@ func (s *PredicateSnapshot) SchedulePodOnAnyNodeMatching(pod *apiv1.Pod, anyNode
 
 // UnschedulePod removes the given Pod from the given Node inside the snapshot.
 func (s *PredicateSnapshot) UnschedulePod(namespace string, podName string, nodeName string) error {
+	if s.draEnabled {
+		nodeInfo, err := s.GetNodeInfo(nodeName)
+		if err != nil {
+			return err
+		}
+
+		var foundPod *apiv1.Pod
+		for _, pod := range nodeInfo.Pods() {
+			if pod.Namespace == namespace && pod.Name == podName {
+				foundPod = pod.Pod
+				break
+			}
+		}
+		if foundPod == nil {
+			return fmt.Errorf("pod %s/%s not found on node %s", namespace, podName, nodeName)
+		}
+
+		if len(foundPod.Spec.ResourceClaims) > 0 {
+			if err := s.ClusterSnapshotStore.DraSnapshot().UnreservePodClaims(foundPod); err != nil {
+				return err
+			}
```

Comment:
https://github.com/kubernetes/autoscaler/pull/7530/#issuecomment-2536807564
> Modifying the state of DRA objects during scheduling simulations (6th commit). This is tied to the ResourceClaim utils, as a lot of them are used here. Is the logic correct? One question that comes to mind - if we remove the last pod reserving a shared claim, should it be deallocated?

Yes, it should be treated as deallocated. Otherwise you cannot move such a pod from one node to another during a simulated scale down. It would be forced to stay on the original node because that is where the claim is allocated.

The rest of the logic looks okay to me.